### PR TITLE
more tipps/tricks for using the repl efficiently

### PR DESCRIPTION
**Pull request by mpollmeier**
fixes https://github.com/ShiftLeftSecurity/product/issues/3470

### CODE REVIEW

File: using-ocular/about/modes.md [hunk 1]
```diff
@@ -2,17 +2,64 @@
 
 ShiftLeft Ocular lets you perform code analysis on CPGs and Security Profiles, both interactively using a REPL or with non-interactive scripts. 
 
-## Using a REPL
+## Using the REPL
 
-A REPL is an interactive shell with support for the Ocular Query Language (OQL). Among other features, a REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support and tab-completion (included in the ShiftLeft Ocular trial version).
+A REPL is an interactive shell with support for the Ocular Query Language (OQL). Among other features, a REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support and tab-completion. It based on Ammonite, so many of it's [documented features](http://ammonite.io/#Features) apply.
```
Comment (khorne3) [inline using-ocular/about/modes.md]:
```suggestion
A REPL is an interactive shell with support for the Ocular Query Language (OQL). Among other features, a REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support, and tab-completion. It is based on Ammonite, so many of its [documented features](http://ammonite.io/#Features) apply.
```

File: using-ocular/about/modes.md [hunk 2]
```diff
@@ -2,17 +2,64 @@
...
 
 The ShiftLeft Ocular underlying shell is an interactive Scala shell which includes useful things like:
 
+* \\&lt;CTRL-c&gt; cancel current operation / clear shell. Does not quit ocular.
```
Comment (khorne3) [inline using-ocular/about/modes.md]:
```suggestion
* \\&lt;CTRL-c&gt; cancel current operation / clear shell. Does not quit Ocular.
```

File: using-ocular/about/modes.md [hunk 3]
```diff
@@ -2,17 +2,64 @@
...
 * \\&lt;TAB&gt; for autocomplete.
-* \\&lt;UP&gt; and <DOWN> for moving through the command history.
-* \\&lt;CTRL-r&gt; to search the command history.
-* `helpMsg` and `status`.
-     
+* \\&lt;UP&gt; and <DOWN> for moving through the command history. Supports multi-line editing!
+* \\&lt;CTRL-LEFT/RIGHT&gt; step through word by word (rather than character by character)
+* \\&lt;CTRL-r&gt; to search the command history. Hit CTRL-r (alternatively UP/DOWN) to cycle through the matches.
```
Comment (khorne3) [inline using-ocular/about/modes.md]: Is CMD+R the command for macOS or is it still CTRL-R?

File: using-ocular/about/modes.md [hunk 4]
```diff
@@ -2,17 +2,64 @@
...
+versionsort.VersionHelper.compare("2.1.0", "2.0.10")
+// res: Int = 1
+
+// scala dependency
+import $ivy.`com.michaelpollmeier::colordiff:0.9`
+colordiff.ColorDiff(List("a", "b"), List("a", "bb"))
+// color coded diff
+```
+
+If the dependencies are not on maven central, you can add a resolver as follows:
```
Comment (khorne3) [inline using-ocular/about/modes.md]:
```suggestion
If the dependencies are not on Maven Central, you can add a resolver as follows:
```

File: using-ocular/about/modes.md (hunk 3 above)
Comment (mpollmeier) [reply using-ocular/about/modes.md]: Good question! I'm guessing it's CTRL-R, but maybe @vlad-alexandru-ionescu can confirm? :bowing_man:

**Comment by mpollmeier** [comment]
Thanks @khorne3 for the good suggestions!

File: using-ocular/about/modes.md (hunk 3 above)
Comment (vlad-alexandru-ionescu) [reply using-ocular/about/modes.md]: It's actually CTRL for both, so we should be good.

File: using-ocular/about/modes.md [hunk 5]
```diff
@@ -2,17 +2,64 @@
...
 
-## Using a REPL
+## Using the REPL
 
-A REPL is an interactive shell with support for the Ocular Query Language (OQL). Among other features, a REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support and tab-completion (included in the ShiftLeft Ocular trial version).
+A REPL is an interactive shell with support for the Ocular Query Language (OQL). Among other features, a REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support, and tab-completion. It is based on Ammonite, so many of its [documented features](http://ammonite.io/#Features) apply.
 
 The ShiftLeft Ocular underlying shell is an interactive Scala shell which includes useful things like:
 
+* \\&lt;CTRL-c&gt; cancel current operation / clear shell. Does not quit Ocular.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/about/modes.md]: CTRL-d does quit however - shall we document it?

Comment (mpollmeier) [reply using-ocular/about/modes.md]: we could try to compete with https://itsfoss.com/how-to-exit-vim/ :)
yeah, good idea, will add

**Pull request merged**